bndtools.core/src/bndtools/launch/JUnitShortcut.java

```diff
@@ -223,18 +287,27 @@ private static boolean isTestable(IType type) throws JavaModelException {
 	/*
 	 * Check if any of the methods has an annotation from the org.junit package
 	 */
+	private static boolean isJUnit(IType z) throws JavaModelException {
+		return isJUnit(z, z.getFlags());
+	}
 
-	private static boolean isJUnit4(IType z) throws JavaModelException {
+	private static boolean isJUnit(IType z, int typeFlags) throws JavaModelException {
 		for (IMethod m : z.getMethods()) {
-			if (Flags.isPublic(m.getFlags())) {
-				for (IAnnotation annotation : m.getAnnotations()) {
-					IType declaringType = m.getDeclaringType();
-					if (declaringType != null) {
-						String[][] names = declaringType.resolveType(annotation.getElementName());
-						if (names != null) {
-							for (String[] pair : names) {
-								if (pair[0].contains("org.junit"))
+			for (IAnnotation annotation : m.getAnnotations()) {
+				IType declaringType = m.getDeclaringType();
+				if (declaringType != null) {
+					String[][] names = declaringType.resolveType(annotation.getElementName());
+					if (names != null) {
+						for (String[] pair : names) {
+							if (pair[0].startsWith("org.junit")) {
+								final int flags = m.getFlags();
+								// Jupiter allows public, protected &
+								// package-protected methods & classes,
+								// JUnit 4 only public
+								if ((pair[0].contains("jupiter") && !Flags.isPrivate(flags))
```

Comment:
Should this not be `contains(".jupiter.")` or `startsWith("org.junit.jupiter.")`?

Reply:
Done.